src/app/frontend/replicasetdetail/updatereplicas_controller.js

```diff
@@ -22,40 +25,67 @@ export default class UpdateReplicasDialogController {
    * @param {!md.$dialog} $mdDialog
    * @param {!angular.$log} $log
    * @param {!ui.router.$state} $state
-   * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
-   * @param {!function(number, function(!backendApi.ReplicaSetSpec),
-   * function(!angular.$http.Response)=)} updateReplicasFn
+   * @param {!angular.$resource} $resource
+   * @param {!angular.$q} $q
+   * @param {string} namespace
+   * @param {string} replicaSet
+   * @param {number} currentPods
+   * @param {number} desiredPods
    * @ngInject
    */
-  constructor($mdDialog, $log, $state, replicaSetDetail, updateReplicasFn) {
+  constructor(
+      $mdDialog, $log, $state, $resource, $q, namespace, replicaSet, currentPods, desiredPods) {
     /** @export {number} */
     this.replicas;
 
-    /** @export {!backendApi.ReplicaSetDetail} */
-    this.replicaSetDetail = replicaSetDetail;
+    /** @export {number} */
+    this.currentPods = currentPods;
+
+    /** @export {number} */
+    this.desiredPods = desiredPods;
+
+    /** @export {string} */
+    this.replicaSet = replicaSet;
+
+    /** @private {string} */
+    this.namespace_ = namespace;
 
     /** @private {!md.$dialog} */
     this.mdDialog_ = $mdDialog;
 
-    /** @private {!function(number, function(!backendApi.ReplicaSetSpec),
-     * function(!angular.$http.Response)=)} */
-    this.updateReplicasFn_ = updateReplicasFn;
-
     /** @private {!angular.$log} */
     this.log_ = $log;
 
     /** @private {!ui.router.$state} */
     this.state_ = $state;
+
+    /** @private {!angular.$resource} */
+    this.resource_ = $resource;
+
+    /** @private {!angular.$q} */
+    this.q_ = $q;
   }
 
   /**
-   * Executes callback function to update replicas count in replica set.
+   * Updates number of replicas in replica set.
+   *
+   * @return {angular.$q.Promise}
    * @export
    */
-  updateReplicasCount() {
-    this.updateReplicasFn_(
-        this.replicas, this.onUpdateReplicasSuccess_.bind(this),
+  updateReplicas() {
+    let resource = getReplicaSetSpecPodsResource(
+        new StateParams(this.namespace_, this.replicaSet), this.resource_);
+    let deferred = this.q_.defer();
+
+    /** @type {!backendApi.ReplicaSetSpec} */
+    let replicaSetSpec = {
+      replicas: this.replicas,
+    };
+
+    resource.save(
+        replicaSetSpec, this.onUpdateReplicasSuccess_.bind(this),
         this.onUpdateReplicasError_.bind(this));
```

Comment:
Or better, see how showDeleteDialog does this. I.e., on successful change the promise is resolved (this would happen here in this.onUpdateReplicasSuccess_)